### PR TITLE
Feature/fixingtheissues

### PR DESCRIPTION
**Pull request by nesrindtanyeri**

### CODE REVIEW

**Review by waynedevilliers — APPROVED** [review]

File: backend/server.js [hunk 1]
```diff
@@ -1,40 +1,25 @@
...
+dotenv.config();
 
 const app = express();
 
-
-const cors = require("cors");
-
-// Allow all origins or specify the frontend's origin
 app.use(cors({
-    origin: "http://localhost:5173" // Your frontend URL
```
Comment (waynedevilliers) [inline backend/server.js:18]: whats this doing?

File: backend/models/leaderboard.model.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import mongoose from "mongoose";
```
Comment (waynedevilliers) [inline backend/models/leaderboard.model.js]: Why do name it leaderboard.model.js. My system does not like the two dots..? Maybe just use leaderboardModel....camelcase

File: backend/routes/leaderboardRoutes.js [hunk 3]
```diff
@@ -1,18 +1,20 @@
-const express = require("express");
```
Comment (waynedevilliers) [inline backend/routes/leaderboardRoutes.js:1]: Use import if:

You're writing modern JavaScript and want to leverage ES Modules.
Your project is configured to use ES Modules (e.g., .mjs or "type": "module" in package.json).

**Review by waynedevilliers — CHANGES_REQUESTED** [review]
please update with main first. I made omse of these changes as well because it was not working yesterday evening

File: backend/server.js (hunk 1 above)
Comment (nesrindtanyeri) [reply backend/server.js:18]: there was an issue, all the data was fetched but not shown on the home page. in order to fix the issue I used cors globally in the server file.

File: backend/models/leaderboard.model.js (hunk 2 above)
Comment (nesrindtanyeri) [reply backend/models/leaderboard.model.js]: I will do this without two dots then.

File: backend/routes/leaderboardRoutes.js (hunk 3 above)
Comment (nesrindtanyeri) [reply backend/routes/leaderboardRoutes.js:1]: there were both import and require.

**Pull request merged**